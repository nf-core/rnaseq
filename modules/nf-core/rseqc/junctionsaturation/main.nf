process RSEQC_JUNCTIONSATURATION {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rseqc:5.0.3--py39hf95cd2a_0' :
        'biocontainers/rseqc:5.0.3--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(bam)
    path  bed

    output:
    tuple val(meta), path("*.pdf"), emit: pdf
    tuple val(meta), path("*.r")  , emit: rscript
    path  "versions.yml"          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    junction_saturation.py \\
        -i $bam \\
        -r $bed \\
        -o $prefix \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rseqc: \$(junction_saturation.py --version | sed -e "s/junction_saturation.py //g")
    END_VERSIONS
    """
}
