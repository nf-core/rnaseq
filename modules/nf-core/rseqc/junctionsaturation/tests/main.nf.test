nextflow_process {

    name "Test Process RSEQC_JUNCTIONSATURATION"
    script "../main.nf"
    process "RSEQC_JUNCTIONSATURATION"
    tag "modules"
    tag "modules_nfcore"
    tag "rseqc"
    tag "rseqc/junctionsaturation"

    test("sarscov2 paired-end [bam]") {

        when {
            process {
                """
                input[0] = [
                            [ id:'test', single_end: false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                           ]
                input[1] = file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.rscript).match("rscript") },
                { assert snapshot(process.out.versions).match("versions") },
                { assert snapshot(file(process.out.pdf[0][1]).name).match("pdf") }
            )
        }

    }

}
