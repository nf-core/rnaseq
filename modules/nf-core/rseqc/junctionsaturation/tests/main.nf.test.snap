{
    "rscript": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.junctionSaturation_plot.r:md5,caa6e63dcb477aabb169882b2f30dadd"
                ]
            ]
        ],
        "timestamp": "2023-11-24T00:15:43.655261555"
    },
    "pdf": {
        "content": [
            "test.junctionSaturation_plot.pdf"
        ],
        "timestamp": "2023-11-24T00:15:43.682673802"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,cd4638541a825dbd44e3cb65d2980aa1"
            ]
        ],
        "timestamp": "2023-12-14T14:59:00.979242265"
    }
}